fe/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -4894,8 +4894,7 @@ public void renameTable(Database db, OlapTable table, TableRenameClause tableRen
 
         // check if rollup has same name
         if (table.getType() == TableType.OLAP) {
```

Comment:
We can just remove the `if (table.getType() == TableType.OLAP)`. The argument is already `OlapTable`